Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -0,0 +1,136 @@
+import UIKit
+import MapboxMaps
+import MapboxCoreNavigation
+
+class SpriteRepository {
+    let imageCache: BimodalImageCache
+    let metadataCache =  SpriteMetaDataCache()
+    var styleURI: StyleURI = .navigationDay
+    var baseURL: String = "https://api.mapbox.com/styles/v1"
+    
+    public var sessionConfiguration: URLSessionConfiguration = URLSessionConfiguration.default {
+        didSet {
+            imageDownloader = ImageDownloader(sessionConfiguration: sessionConfiguration)
+        }
+    }
+
+    public static let shared = SpriteRepository.init()
+    fileprivate(set) var imageDownloader: ReentrantImageDownloader
+    
+    init(imageCache: BimodalImageCache = ImageCache(), withDownloader downloader: ReentrantImageDownloader = ImageDownloader()) {
+        self.imageCache = imageCache
+        self.imageDownloader = downloader
+    }
+    
+    func updateRepository(updateStyleURI: StyleURI? = nil, updateBaseURL: String? = nil, completion: @escaping CompletionHandler) {
+        let styleURI = updateStyleURI ?? styleURI
+        let baseURL = updateBaseURL ?? baseURL
+        
+        guard let styleID = styleURI.rawValue.components(separatedBy: "styles")[safe: 1],
+              let accessToken = NavigationSettings.shared.directions.credentials.accessToken,
+              let spriteRequestURL = URL(string: baseURL + styleID + "/sprite@2x.png?access_token=" + accessToken),
+              let metadataRequestURL = URL(string: baseURL + styleID + "/sprite@2x?access_token=" + accessToken) else {
+                  return
+              }
```

Comment:
Let's use [URLComponents](https://developer.apple.com/documentation/foundation/urlcomponents) for URL generation.

Reply:
Thanks for the suggestion, I'll replace it.